File: Binary Tree/Is node present/IsNodePresent.cpp
#include <iostream>
#include <queue>
using namespace std;
template <typename T>
class TreeNode
{
public:
	TreeNode<T> *left;
	TreeNode<T> *right;
	T data;
	TreeNode(T data)
	{
		this->data=data;
	}
};
TreeNode<int> * takeInput()
{
	queue<TreeNode<int>*> input;
	int data;
	cin>>data;
	TreeNode<int> *root=new TreeNode<int>(data);
	input.push(root);
	while(!input.empty())
	{
		TreeNode<int>*temp=input.front();
		input.pop();
		cin>>data;
		if (data!=-1)
		{
			temp->left=new TreeNode<int>(data);
			input.push(temp->left);
		}else{
			temp->left=NULL;
		}
		cin>>data;
		if (data!=-1)
		{
			temp->right=new TreeNode<int>(data);
			input.push(temp->right);
		}else{
			temp->right=NULL;
		}
	}
	return root;
}
void printTree(TreeNode<int>*root)
{
	if (!root)
	{
		return;
	}
	cout<<root->data<<" : ";
	if (root->left)
	{
		cout<<" L "<<root->left->data<<" , ";
	}
	if (root->right)
	{
		cout<<" R "<<root->right->data;
	}
	cout<<endl;
	printTree(root->left);
	printTree(root->right);
}
bool checkPresent(TreeNode<int>* root  ,int data)
{
	if (!root)
	{
		return false;
	}
	if (root->data==data)
	{
		return true;
	}
	return (checkPresent(root->left  , data) || checkPresent(root->right  , data));
}
int main()
{
	TreeNode<int>*root;
	root=takeInput();
	printTree(root);
	int data;
	cin>>data;
	bool isPresent=checkPresent(root  , data);
	if (isPresent)
	{
		cout<<"true";
	}else{
		cout<<"false";
	}
}

File: Binary Tree/Is node present/IsNodePresent.java
import java.util.Scanner;
import java.util.Queue;
import java.util.LinkedList;
class TreeNode<T>
{
	public TreeNode<T> left;
	public TreeNode<T> right;
	public T data;
	TreeNode(T data)
	{
		this.data=data;
	}
}
class IsNodePresent
{
	public static TreeNode<Integer> takeInput()
	{
		Scanner scan=new Scanner(System.in);
		Queue<TreeNode<Integer> > input=new LinkedList<TreeNode<Integer>>();
		TreeNode<Integer>root;
		int data=scan.nextInt();
		root=new TreeNode<Integer>(data);
		input.add(root);
		while(!input.isEmpty())
		{
			TreeNode<Integer> temp=input.remove();
			//root of left subtree
			data=scan.nextInt();
			if (data!=-1) {
				temp.left=new TreeNode<Integer>(data);
				input.add(temp.left);
			}else{
				temp.left=null;
			}
			data=scan.nextInt();
			if (data!=-1) {
				temp.right=new TreeNode<Integer>(data);
				input.add(temp.right);
			}else{
				temp.right=null;
			}
		}
		return root;
	}
	public static boolean checkPresent(TreeNode<Integer> root , int data)
	{
		if (root==null) {
			return false;
		}
		if (root.data==data) {
			return true;
		}
		return (checkPresent(root.left , data) || checkPresent(root.right , data));
	}
	public static void printTree(TreeNode<Integer> root)
	{
		if (root==null) {
			return;
		}
		System.out.print(root.data+" : ");
		if (root.left!=null) {
			System.out.print(" L "+root.left.data+" , ");
		}
		if (root.right!=null) {
			System.out.print(" R "+root.right.data);
		}
		System.out.println();
		printTree(root.left);
		printTree(root.right);
	}
	public static void main(String[] args) {
		TreeNode<Integer> root=takeInput();
		printTree(root);
		Scanner scan=new Scanner(System.in);
		int data=scan.nextInt();
		boolean isPresent=checkPresent(root , data);
		if (isPresent) {
			System.out.println("true");
		}else{
			System.out.println("false");
		}
	}
}